controllers/datadogagent/merger/volume.go

```diff
@@ -59,13 +65,18 @@ func (impl *volumeManagerImpl) AddVolumeWithMergeFunc(volume *corev1.Volume, vol
 	return nil
 }
 
-func (impl *volumeManagerImpl) AddVolumeToContainerWithMergeFunc(volume *corev1.Volume, volumeMount *corev1.VolumeMount, containerName commonv1.AgentContainerName, volumeMergeFunc VolumeMergeFunction, volumeMountMergeFunc VolumeMountMergeFunction) error {
+func (impl *volumeManagerImpl) AddVolumeToContainersWithMergeFunc(volume *corev1.Volume, volumeMount *corev1.VolumeMount, containerNames []commonv1.AgentContainerName, volumeMergeFunc VolumeMergeFunction, volumeMountMergeFunc VolumeMountMergeFunction) error {
```

Comment:
A solution should be to call the function `AddVolumeToContainerWithMergeFunc()` in a loop on `containerNames`. It will not be optimum in term of tentative to add the `Volume`, but it can reduce the code

Reply:
yea i thought about that, but in the worst case scenario it would loop through the `impl.podTmpl.Spec.Containers` 4 x 5 times..

we could add the `return` into the `if` block which would help a bit. do you prefer this route?

Reply:
we can keep the function as it is too 👍